src/monster.h

```diff
@@ -19,6 +19,19 @@ using mtype_id = string_id<mtype>;
 
 typedef std::map< mfaction_id, std::set< int > > mfactions;
 
+class mon_special_attack : public JsonSerializer {
+public:
+    int cooldown;
+    bool enabled;
+
+    // FIXME optional parameters are stand-in for better syntax in savegame_json.cpp for creating new entries
+    mon_special_attack(int c = 0, bool e = true) : cooldown(c), enabled(e) {};
+
+    using JsonSerializer::serialize;
+    virtual void serialize(JsonOut &jsout) const override;
```

Comment:
*Why* is there no matching deserialize function for this? This looks very strange and confusing.